pkg/storage/s3.go

```diff
@@ -452,6 +453,9 @@ type s3ObjectReader struct {
 	name    string
 	reader  io.ReadCloser
 	pos     int64
+	size    *int64
```

Comment:
could we parse the Content-Range to retrieve the file size for ranged request (and use Content-Length for non-ranged request)? to save an extra HEAD request.

according to https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html#API_GetObject_Example_11, if we issue the GET request with

```
            Range: bytes=0-9
```

the reply will contain

```
            Accept-Ranges: bytes
            Content-Range: bytes 0-9/443
            Content-Type: text/plain
            Content-Length: 10
```

so that "433" should be the actual file size.

Reply:
I will do some test for this

Reply:
In my test, the `AcceptRanges` field return by minio is always `bytes`, not sure if s3 will return the right value